File: Backend/PC4MT/Frontend/src/app/maintenance/viewmodels/downloadSettingsViewModel.ts
export class DownloadSettignsViewModel {
  public maxSentencesPerDownload: number | null;
  public maxSentencesPerUser: number | null;
  public maxSentencesL1L2: number | null;
  public maxDownloadsPerUser: number | null;

  constructor(registo?: DownloadSettignsViewModel) {
    this.copyFrom(registo);
  }

  public copyFrom(registo?: DownloadSettignsViewModel): void {
    if (registo) {
      this.maxSentencesPerDownload = registo.maxSentencesPerDownload;
      this.maxSentencesPerUser = registo.maxSentencesPerUser;
      this.maxSentencesL1L2 = registo.maxSentencesL1L2;
      this.maxDownloadsPerUser = registo.maxDownloadsPerUser;
    }
  }
}

File: PC4MT.BLL/ViewModels/DownloadSettingsViewModel.cs
using PC4MT.DAL.Data.Models;

namespace PC4MT.BLL.ViewModels
{
    public class DownloadSettingsViewModel
    {
        public int? MaxSentencesPerDownload { get; set; }

        public int? MaxSentencesPerUser { get; set; }

        public int? MaxSentencesL1L2 { get; set; }

        public int? MaxDownloadsPerUser { get; set; }

        public static DownloadSettingsViewModel MapEntityToViewModel(ConfigDownload entity)
        {
            if (entity == null)
            {
                return null;
            }

            return new DownloadSettingsViewModel()
            {
                MaxSentencesPerDownload = entity.MaxSentencesPerDownload,
                MaxSentencesPerUser = entity.MaxSentencesPerUser,
                MaxSentencesL1L2 = entity.MaxSentencesL1L2,
                MaxDownloadsPerUser = entity.MaxDownloadsPerUser
            };
        }

        public static ConfigDownload MapViewModelToEntity(DownloadSettingsViewModel viewModel)
        {
            if (viewModel == null)
            {
                return null;
            }

            return new ConfigDownload()
            {
                MaxSentencesPerDownload = viewModel.MaxSentencesPerDownload,
                MaxSentencesPerUser = viewModel.MaxSentencesPerUser,
                MaxSentencesL1L2 = viewModel.MaxSentencesL1L2,
                MaxDownloadsPerUser = viewModel.MaxDownloadsPerUser
            };
        }
    }
}
